### ISSUE TITLE
Syntax error with pk:"autoincr"

### ISSUE THREAD

**Issue by Alexendoo** [src-d/go-kallax#158]
`kallax version 1.2.1`
`go version go1.8 windows/amd64`

The following code:

```go
package main

import (
	"database/sql"
	"fmt"

	_ "github.com/lib/pq"
	"gopkg.in/src-d/go-kallax.v1"
)

//go:generate kallax gen

type Foo struct {
	kallax.Model
	ID int64 `pk:"autoincr"`
}

func main() {
	db, _ := sql.Open("postgres", "postgres://devel:devel@localhost/foo?sslmode=disable")

	store := NewFooStore(db)

	err := store.Insert(NewFoo())

	fmt.Printf("%+#v\n", err)
}
```

generates the error:

```go
&pq.Error{
	Severity: "ERROR",
	Code: "42601",
	Message: "syntax error at or near \")\"",
	Detail: "",
	Hint: "",
	Position: "25",
	InternalPosition: "",
	InternalQuery: "",
	Where: "",
	Schema: "",
	Table: "",
	Column: "",
	DataTypeName: "",
	Constraint: "",
	File: "scan.l",
	Line: "1087",
	Routine: "scanner_yyerror",
}
```

Without `autoincr` there is no error

**Comment by erizocosmico** [comment]
Can you copy the output of using `store.Debug().Insert(NewFoo())`?
On Mon, 15 May 2017 at 17:32, Alex Macleod <notifications@github.com<mailto:notifications@github.com>> wrote:

kallax version 1.2.1
go version go1.8 windows/amd64

The following code:

package main

import (
        "database/sql"
        "fmt"

        _ "github.com/lib/pq<http://github.com/lib/pq>"
        "gopkg.in/src-d/go-kallax.v1<http://gopkg.in/src-d/go-kallax.v1>"
)

//go:generate kallax gen

type Foo struct {
        kallax.Model
        ID int64 `pk:"autoincr"`
}

func main() {
        db, _ := sql.Open("postgres", "postgres://devel:devel@localhost/foo?sslmode=disable")

        store := NewFooStore(db)

        err := store.Insert(NewFoo())

        fmt.Printf("%+#v\n", err)
}

generates the error:

&pq.Error{
        Severity: "ERROR",
        Code: "42601",
        Message: "syntax error at or near \")\"",
        Detail: "",
        Hint: "",
        Position: "25",
        InternalPosition: "",
        InternalQuery: "",
        Where: "",
        Schema: "",
        Table: "",
        Column: "",
        DataTypeName: "",
        Constraint: "",
        File: "scan.l",
        Line: "1087",
        Routine: "scanner_yyerror",
}

Without autoincr there is no error

**Comment by Alexendoo** [comment]
It doesn't provide any output, I've noticed that debug `Insert`s don't seem to provide any output even when they work though so perhaps an unrelated issue

**Comment by erizocosmico** [comment]
You model has nothing but the pk, and generates the following SQL `INSERT INTO foo VALUES ()`, which is invalid.

We'll catch this error and return a specific kallax error for this case. Also, I will fix the lack of logging for inserts.

**Comment by erizocosmico** [comment]
Fixed in `1.2.2`.
